### ISSUE TITLE
Set a node name in the 'Create New Node' dialog box

### ISSUE THREAD

**Issue by chucklepie** [godotengine/godot-proposals#1686]
**Describe the project you are working on:**
It's a beautiful little 2D retro remake of a class ZX Spectrum game called Rex, from 1989. Check it out :)

**Describe the problem or limitation you are having in your project:**
I'd say about 80% of the time I add a node I then go into the scene pane and rename it. It's a tiny bit tedious

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
I did check first! Simply add a text box to the create new node dialog, auto-filling with the default, called 'node name' allowing it to be done here. Then you don't have to go to the scene and select rename then type.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Please accept my apology, I couldn't be bothered to add a font to my pixel editor ;-)
![image](https://user-images.githubusercontent.com/12863685/96368854-566c6e00-114e-11eb-8612-a3bd228c8ad4.png)

**Is there a reason why this should be core and not an add-on in the asset library?:**
It's a quality of life change, one that will speed up things and is a simple addition

**Comment by KoBeWi** [comment]
You can press enter after adding node to rename it. The catch is that you can't use mouse.
![RVxtc6Z0WR](https://user-images.githubusercontent.com/2223172/96453542-e7a91680-121a-11eb-8adb-ac814babf2d5.gif)

**Comment by chucklepie** [comment]
I my solution better 😉

It matches how other dialogs do it, eg signal connect

**Comment by RoniPerson** [comment]
As mentioned in #3568 I would add the addition that when typing something in the searchfield that dos not match any node name it gets copied to the name field automatically, preventing one copy and paste in this case.

**Comment by dalexeev** [comment]
> I would add the addition that when typing something in the searchfield that dos not match any node name it gets copied to the name field automatically

I think you'll get used to using the correct field. :slightly_smiling_face: 

We probably need to swap the fields so that the Name is first and does not separate the Filter from the Matches:

![](https://user-images.githubusercontent.com/47700418/142447848-da312867-f9bc-4d8d-867c-a778db234479.png)

**Comment by RoniPerson** [comment]
> 
> 
> > I would add the addition that when typing something in the searchfield that dos not match any node name it gets copied to the name field automatically
> 
> I think you'll get used to using the correct field. 🙂
> 
> We probably need to swap the fields so that the Name is first and does not separate the Filter from the Matches:

I have to say that I liked the mockup with both fields next to each better:
![](https://user-images.githubusercontent.com/47700418/142419768-2d9912a2-a11d-4709-8910-308169377b78.png)

It keeps the interface simpler and nodenames are not so lang that the smaller search bar would overflow.

**Comment by LightningAA** [comment]
> I my solution better 😉 It matches how other dialogs do it, eg signal connect

The workflow this proposal suggests is you select the node you want, press tab/click to select the name dialog, enter the name and then hit enter to confirm. With the current workflow, you select the node you want, hit enter, then just hit enter a second time if you want to rename it. In my opinion the current system is less complex and already works better. I'm not strictly against this proposal, but I feel like this is trying to solve a problem that has already been solved.

**Comment by spindlebink** [comment]
Per [my comment here](https://github.com/godotengine/godot-proposals/issues/3568#issuecomment-974240764), what about forgoing the dedicated box and instead allowing separation of node type and name with a colon? Doing it this way would turn it into an easily-accessible, minor change to the current workflow which would enable the more keyboard-centric `n2d:NodeName` shortcut. It also shouldn't require much modification of the current code.

This method also fits with the convenience-oriented nature of the node adding screen--I kind of agree with @LightningAA here and feel like a dedicated box for the name isn't that much more straightforward.

**Comment by RoniPerson** [comment]
> This method also fits with the convenience-oriented nature of the node adding screen--I kind of agree with @LightningAA here and feel like a dedicated box for the name isn't that much better than what we currently have.

Yeah I think you are right here though I'm not sure if `type:name` or `name:type` would be better. 
Also solving issues over some sort of string pattern always feels a little bit hacky for me. But it would at least help solving the problem without huge changes to the ui.

**Comment by spindlebink** [comment]
> Yeah I think you are right here though I'm not sure if `type:name` or `name:type` would be better

A theoretical `type:name` pattern would fit with the current workflow and make the convenience name optional. To get the current search functionality with `name:type` format you'd need to prefix every search with `:` which would be icky.

> feels a little bit hacky for me

Think of it like a hotkey, or like the meta+P go-to-symbol palette that's big in code editors these days.
